tests/scripts/thread-cert/Cert_5_2_05_AddressQuery.py

```diff
@@ -30,110 +30,130 @@
-        self.nodes[REED].add_whitelist(self.nodes[ROUTER2].get_addr64())
-        self.nodes[REED].set_router_upgrade_threshold(0)
-        self.nodes[REED].enable_whitelist()
-
-        self.nodes[ROUTER2].set_panid(0xface)
-        self.nodes[ROUTER2].set_mode('rsdn')
-        self.nodes[ROUTER2].add_whitelist(self.nodes[LEADER].get_addr64())
-        self.nodes[ROUTER2].add_whitelist(self.nodes[REED].get_addr64())
-        self.nodes[ROUTER2].add_whitelist(self.nodes[ED2].get_addr64())
-        self.nodes[ROUTER2].add_whitelist(self.nodes[ED3].get_addr64())
-        self.nodes[ROUTER2].enable_whitelist()
-        self.nodes[ROUTER2].set_router_selection_jitter(1)
-
-        self.nodes[ED2].set_panid(0xface)
-        self.nodes[ED2].set_mode('rsn')
-        self.nodes[ED2].add_whitelist(self.nodes[ROUTER2].get_addr64())
-        self.nodes[ED2].enable_whitelist()
-
-        self.nodes[ED3].set_panid(0xface)
-        self.nodes[ED3].set_mode('rsn')
-        self.nodes[ED3].add_whitelist(self.nodes[ROUTER2].get_addr64())
-        self.nodes[ED3].enable_whitelist()
+        self.nodes[DUT_REED].set_panid(0xface)
+        self.nodes[DUT_REED].set_mode('rsdn')
+        self.nodes[DUT_REED].add_whitelist(self.nodes[ROUTER1].get_addr64())
+        self.nodes[ROUTER1].add_whitelist(self.nodes[DUT_REED].get_addr64())
+        self.nodes[DUT_REED].set_router_upgrade_threshold(0)
+        self.nodes[DUT_REED].enable_whitelist()
+        self.nodes[DUT_REED].set_router_selection_jitter(1)
+
+        self.sniffer = config.create_default_thread_sniffer(SNIFFER)
+        self.sniffer.start()
 
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
+        # 1. LEADER: Configure the Leader to be a DHCPv6  Border Router for prefix 2001::.
         self.nodes[LEADER].start()
         self.nodes[LEADER].set_state('leader')
         self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+        self.nodes[LEADER].add_prefix('2001::/64', 'pdros')
+        self.nodes[LEADER].register_netdata()
 
-        self.nodes[BR1].start()
+        # 2. BR: Attach the Border_Router to the network and configure On-Mesh Prefix.
+        self.nodes[BR].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[BR1].get_state(), 'router')
-
-        self.nodes[BR1].add_prefix('2001:2:0:3::/64', 'paros')
-        self.nodes[BR1].add_prefix('2001:2:0:4::/64', 'paros')
-        self.nodes[BR1].register_netdata()
+        self.assertEqual(self.nodes[BR].get_state(), 'router')
+        self.nodes[BR].add_prefix('2002::/64', 'paros')
+        self.nodes[BR].register_netdata()
+
+        # 3. All: Ensure topology is formed correctly without the DUT_REED.
+        for i in range(2, 17):
+            if i == BR:
+                continue
+            self.nodes[i].start()
+            time.sleep(5)
```

Comment:
Sleep for each node spends 14 * 5s. How about sleeping 5s after staring all nodes?

Reply:
no. it can't work in that way.